### PR TITLE
Add vm template and host integartion test

### PR DESCRIPTION
**Pull request by guangbochen**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

need to add host and VM template integration tests

**Solution:**
add host and VM template integration tests

**Related Issue:**
https://github.com/rancher/harvester/issues/388

**Test plan:**

N/A

### CODE REVIEW

**Comment by guangbochen** [comment]
@gitlawr updated, thanks

File: tests/framework/helper/http.go [hunk 1]
```diff
@@ -11,6 +11,8 @@ import (
 	"sigs.k8s.io/yaml"
 
 	"github.com/rancher/harvester/pkg/config"
+
+	"github.com/rancher/apiserver/pkg/types"
```
Comment (gitlawr) [inline tests/framework/helper/http.go]: format the import

File: tests/integration/api/vm_template_apis_test.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+				templateVersion.Spec.TemplateID = templateID
+				templateVersion.Spec.VM = NewDefaultTestVMBuilder().vm.Spec
+
+				respCode, respBody, err := helper.PostObjectByYAML(templateVersionAPI, templateVersion)
+				MustRespCodeIs(http.StatusCreated, "create template version", err, respCode, respBody)
+
+			})
+
+			// sleep to wait for template controller to sync the default version id
```
Comment (gitlawr) [inline tests/integration/api/vm_template_apis_test.go]: remove this comment

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**